libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/OAuth2.java

```diff
@@ -387,17 +387,13 @@ public static URI getFrontdoorUrl(URI url,
     }
 
     /**
-     * Computes the scope parameter from an array of scopes. Also adds
-     * the 'refresh_token' scope if it hasn't already been added.
+     * Computes the scope parameter from an array of scopes.
      *
      * @param scopes Array of scopes.
-     * @return Scope parameter.
+     * @return Scope parameter string (possibly empty).
      */
     public static String computeScopeParameter(String[] scopes) {
```

Comment:
This is an important change for this PR.
`ScopeParser.computeScopeParameter(scopes)` will leave an empty scopes list unchanged.